pkg/prometheus/promcfg.go

```diff
@@ -2290,6 +2291,15 @@ func (cg *ConfigGenerator) generateScrapeConfig(
 	relabelings := initRelabelings()
 	labeler := namespacelabeler.New(cpf.EnforcedNamespaceLabel, cpf.ExcludedFromEnforcement, false)
 
+	if sc.Spec.JobName != "" {
+		relabelings = append(relabelings, yaml.MapSlice{
+			{Key: "target_label", Value: "scrapeJob"},
+			{Key: "action", Value: "replace"},
+			{Key: "replacement", Value: sc.Spec.JobName},
+		})
+		cfg = append(cfg, yaml.MapItem{Key: "relabel_configs", Value: relabelings})
```

Comment:
```suggestion
```

Reply:
If I remove this line (just copied from the `if sc.Spec.RelabelConfigs != nil {` statement) the relabel_config isn't applied anymore. Is there a different trick I missed?